### PR TITLE
fix(csi): wait until messaging channel is established before flushing

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by erwinmombay** [comment]
@dvoytenko PTAL

File: src/service/performance-impl.js [hunk 1]
```diff
@@ -109,22 +112,29 @@ export class Performance {
...
-    this.setDocumentInfoParams_();
-
-    // forward all queued ticks to the viewer.
-    this.flushQueuedTicks_();
-    // We need to call flush right away in case the viewer is available
-    // later than the amp codebase had invoked the performance services'
-    // `flush` method to forward ticks.
-    this.flush();
+
+    return this.viewer_.whenMessagingReady().then(() => {
```
Comment (dvoytenko) [inline src/service/performance-impl.js]: It's nullable. And modify/check `isPerformanceTrackingOn` as well.

File: src/service/viewer-impl.js [hunk 2]
```diff
@@ -944,6 +944,14 @@ export class Viewer {
       }
     });
   }
+
+  /**
+   * Resolves when there is a messaging channel established with the viewer.
+   * @return {!Promise}
```
Comment (dvoytenko) [inline src/service/viewer-impl.js]: nullable?

File: src/service/performance-impl.js (hunk 1 above)
Comment (erwinmombay) [inline src/service/performance-impl.js]: done.

File: src/service/viewer-impl.js (hunk 2 above)
Comment (erwinmombay) [inline src/service/viewer-impl.js]: done.

**Comment by erwinmombay** [comment]
@dvoytenko PTAL

File: src/service/viewer-impl.js [hunk 3]
```diff
@@ -431,10 +431,15 @@ export class Viewer {
...
+   * Will also return false if no messaging channel is established, this
+   * means the AMP page is not embedded.
    * @return {boolean}
    */
   isPerformanceTrackingOn() {
-    return this.performanceTracking_;
+    // If there is no messagingMaybePromise_, then document is not
+    // embedded and no performance tracking is needed since there is nobody
+    // to forward the events.
+    return this.performanceTracking_ && this.messagingMaybePromise_;
```
Comment (dvoytenko) [inline src/service/viewer-impl.js]: `!!this.messagingMaybePromise_` to keep it boolean

**Comment by dvoytenko** [comment]
LGTM with one comment.

File: src/service/viewer-impl.js (hunk 3 above)
Comment (erwinmombay) [inline src/service/viewer-impl.js]: done.

**Pull request merged**